gradle/scripts/lib/kotlin.gradle

```diff
@@ -27,7 +25,7 @@ configure(projectsWithFlags('kotlin')) {
             disabledRules = ["import-ordering"]
 
             filter {
-                exclude("**/gen-src/**")
+                exclude { it.file.path.contains("gen-src/") }
```

Comment:
Minor Question) What kind of patterns does the updated rule filter as opposed to the original pattern?

Reply:
Not sure of the exact reason but the project I've recently setup won't build well without adding this directive.
It tries to lint the files in `project-root/gen-src/main/kotlinGrpc/*`.

The `ktlint` plugin states that this can happen for dynamically attaches sources that are located outside the project dir (ref. https://github.com/JLLeitschuh/ktlint-gradle#faq).

I will make a reproducer and share it here.

Reply:
@jrhee17 @minwoox 
Sorry I was late 😅. Made a reproducer at https://github.com/ks-yim/gradle-scripts-grpc-kotlin-ktlint-error-reproducer.
The build fails with the following error.

```shell
> Task :grpc-app:compileJava
Note: /Users/ks-yim/IdeaProjects/gradle-scripts-ktlint/grpc-app/gen-src/main/java/com/google/protobuf/compiler/PluginProtos.java uses or overrides a deprecated API.
Note: Recompile with -Xlint:deprecation for details.

> Task :grpc-app:ktlintMainSourceSetCheck FAILED
/Users/ks-yim/IdeaProjects/gradle-scripts-ktlint/grpc-app/gen-src/main/kotlinGrpc/HelloGrpcKt.kt:1:1 object HelloServiceGrpcKt should be declared in a file named HelloServiceGrpcKt.kt (cannot be auto-corrected) (filename)
/Users/ks-yim/IdeaProjects/gradle-scripts-ktlint/grpc-app/gen-src/main/kotlinGrpc/HelloGrpcKt.kt:27:1 Unexpected indentation (2) (should be 4) (indent)
/Users/ks-yim/IdeaProjects/gradle-scripts-ktlint/grpc-app/gen-src/main/kotlinGrpc/HelloGrpcKt.kt:29:1 Unexpected indentation (2) (should be 4) (indent)
/Users/ks-yim/IdeaProjects/gradle-scripts-ktlint/grpc-app/gen-src/main/kotlinGrpc/HelloGrpcKt.kt:30:1 Unexpected indentation (2) (should be 4) (indent)
/Users/ks-yim/IdeaProjects/gradle-scripts-ktlint/grpc-app/gen-src/main/kotlinGrpc/HelloGrpcKt.kt:31:1 Unexpected indentation (4) (should be 8) (indent)
/Users/ks-yim/IdeaProjects/gradle-scripts-ktlint/grpc-app/gen-src/main/kotlinGrpc/HelloGrpcKt.kt:33:1 Unexpected indentation (2) (should be 4) (indent)
/Users/ks-yim/IdeaProjects/gradle-scripts-ktlint/grpc-app/gen-src/main/kotlinGrpc/HelloGrpcKt.kt:34:1 Unexpected indentation (4) (should be 8) (indent)
/Users/ks-yim/IdeaProjects/gradle-scripts-ktlint/grpc-app/gen-src/main/kotlinGrpc/HelloGrpcKt.kt:35:1 Unexpected indentation (4) (should be 8) (indent)
...
```